src/lib/shim/patch_vdso.c

```diff
@@ -164,13 +164,17 @@ static void _inject_trampoline(struct ParsedElf* parsedElf, const char* vdsoFnNa
         warning("Couldn't find symbol '%s' to override", vdsoFnName);
         return;
     }
-    // Loose upper bound on size of instructions we're writing, below.
-    if (symbol->st_size < 20) {
+
+    // Manually calculated trampoline size. Later validated in assertion.
+    const size_t trampolineSize = 13;
+
+    if (symbol->st_size < trampolineSize) {
         warning("Symbol '%s' not large enough to inject trampoline", vdsoFnName);
         return;
     }
```

Comment:
Should this be a failure instead? I feel like shim warnings will rarely be noticed, and this could cause the sim to produce unexpected results.

Reply:
Good point. Done.